src/main/kotlin/com/cosmotech/api/containerregistry/ContainerRegistryService.kt

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+package com.cosmotech.api.containerregistry
+
+import com.cosmotech.api.config.CsmPlatformProperties
+import com.cosmotech.api.exceptions.CsmResourceNotFoundException
+import java.util.Base64
+import org.apache.http.client.methods.HttpUriRequest
+import org.apache.http.client.methods.RequestBuilder
+import org.apache.http.impl.client.HttpClients
+import org.apache.http.util.EntityUtils
+import org.slf4j.LoggerFactory
+import org.springframework.http.HttpHeaders
+import org.springframework.stereotype.Service
+
+@Service("csmContainerRegistry")
+class ContainerRegistryService(private val csmPlatformProperties: CsmPlatformProperties) {
```

Comment:
As you remove the mechanism for azure, using specific client, does it works exactly the same with an azure container registry?

Reply:
I tested with azure container registry, it works